user/languages/android.md

```diff
@@ -18,99 +18,71 @@ Android builds are not available on the macOS environment.
 
 Travis CI environment provides a large set of build tools for JVM languages with [multiple JDKs, Ant, Gradle, Maven](/user/languages/java/#overview), [sbt](/user/languages/scala#projects-using-sbt) and [Leiningen](/user/languages/clojure).
 
-By setting
+Here is an example `.travis.yml` for an Android project without emulator configuration:
 
 ```yaml
-language: android
-dist: trusty
-```
-{: data-file=".travis.yml"}
+dist: focal
+language: java
+jdk: openjdk11
+
+env:
+  global:
+  # Android command line tools, check for updates here https://developer.android.com/studio/#command-tools
+  - COMMAND_LINE_TOOLS_VERSION=7583922
+  - ANDROID_HOME=$HOME/android-sdk
+  # The build tools version used by your project
+  - BUILD_TOOLS_VERSION=30.0.3
+  # The target SDK version used by your project
+  - TARGET_SDK_VERSION=31
 
-in your `.travis.yml` file, your project will be built in the Android environment which provides [Android SDK Tools](http://developer.android.com/tools/sdk/tools-notes.html) 25.2.3.
+before_cache:
+  # Do not cache a few Gradle files/directories (see https://docs.travis-ci.com/user/languages/java/#caching)
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 
-> Android builds are only supported on our Trusty image at this time hence you'll need to explicitly specify `dist: trusty` in your .travis.yml file.
+cache:
+  directories:
+    # Android SDK
+    - $HOME/android-cmdline-tools
+    - $HOME/android-sdk
 
-Here is an example `.travis.yml` for an Android project:
+    # Gradle dependencies
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
 
-```yaml
```

Comment:
@johnjohndoe I think we want to keep the old 'Trusty-based' recipe in .travis.yml for reference below your no-emulator-required example as a separate section. Would you mind editing the file to keep it as a separate section?

And many thanks for contributing!

Reply:
``` yaml